linera-chain/src/certificate/generic.rs

```diff
@@ -139,12 +132,13 @@ impl<T: Clone> Clone for GenericCertificate<T> {
 }
 
 #[cfg(with_testing)]
-impl<T: Eq + PartialEq> Eq for GenericCertificate<T> {}
+impl<T: CertificateValue + Eq + PartialEq> Eq for GenericCertificate<T> {}
 #[cfg(with_testing)]
-impl<T: Eq + PartialEq> PartialEq for GenericCertificate<T> {
+impl<T: CertificateValue + Eq + PartialEq> PartialEq for GenericCertificate<T> {
     fn eq(&self, other: &Self) -> bool {
         self.hash() == other.hash()
             && self.round == other.round
             && self.signatures == other.signatures
+        // && self::KIND == other::KIND
```

Comment:
Does this comment mean that the type system already implies that the kinds match?

Reply:
Removed in [RM commented-out line.](https://github.com/linera-io/linera-protocol/pull/3701/commits/3b22e9f6e34afdf2199e22e8809bcf925a5c4fd7)